core/templates/filters/translate.pipe.spec.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the translate pipe.
+ */
+
+import { async, TestBed } from '@angular/core/testing';
+import { EventEmitter, ChangeDetectorRef } from '@angular/core';
+
+import { TranslateService } from 'services/translate.service';
+import { TranslatePipe } from './translate.pipe';
+import { UtilsService } from 'services/utils.service';
+
+
+class MockChangeDetectorRef {
+  markForCheck() {
+    return true;
+  }
+}
+
+class MockTranslateService {
```

Comment:
Why are we mocking all of the translate service, rather than just the parts which need to be mocked? Better to do just the latter, to reduce skew.

Reply:
The translate pipe has changed a lot over the reviews, so indeed, this mocked service has more than what is needed. Removed the extra and I guess this was an isolated incident, but I see another comment. I think there is no other mock that has more than what is needed.

Reply:
Ah, I see. Yes, then I think your approach sounds good, thank you for the pointers!

Would you mind updating the frontend test wiki doc accordingly so that we can use it as a reference?

Reply:
Ok, I will do that